roles/mongodb-server/tasks/primary.yml

```diff
@@ -0,0 +1,18 @@
+---
+- name: initiate replica set
+  command: mongo --host {{ endpoints.mongodb }} --eval 'rs.initiate()'
+
+- name: wait for replica set init to complete
+  pause: seconds=20
+
+- name: rename primary replica member
+  command: mongo --host {{ endpoints.mongodb }} --eval 'cfg=rs.conf(); cfg.members[0].host="{{ primary_ip }}:{{ mongodb.port }}"; rs.reconfig(cfg)'
+
+- name: install pymongo for mongodb_user module
+  pip: name=pymongo state=present
+
+- name: create admin user
+  mongodb_user: database=admin name=admin password={{ mongodb.ceilometer_password }} roles='userAdminAnyDatabase' state=present
+
+- name: create ceilometer user
```

Comment:
It should be done from ceilometer as ceilometer depends on mongo being present.  If, god forbid, there were another place we wanted to use mongo, we would not necessarily want the ceilometer user present.